### PR TITLE
Build alpine Docker images in CI when linux/amd64 is in platform list

### LINKED ISSUE #1140: alpine Docker images are no longer being published
The last alpine Docker images to be published were for 0.7.0. I broke this in my arm64 Docker image changes in #1099. Just creating this issue to reference from the PR I'm about to open.

### PR DESCRIPTION
**Pull request by cap10morgan**
- [X] I have read the [developer documentation](https://github.com/babashka/babashka/blob/master/doc/dev.md).

- [X] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code). Fixes #1140 

- [ ] This PR contains a [test](https://github.com/babashka/babashka/blob/master/doc/dev.md#tests) to prevent against future regressions - N/A

- [X] I have updated the [CHANGELOG.md](https://github.com/babashka/babashka/blob/master/CHANGELOG.md) file with a description of the addressed issue.

### CODE REVIEW

File: .circleci/script/docker [hunk 1]
```diff
@@ -44,9 +44,9 @@ if [ -z "$CIRCLE_PULL_REQUEST" ] && [ "$CIRCLE_BRANCH" = "master" ]; then
     done
 
     # alpine doesn't provide upstream arm64 images yet
-    if [[ $platform == "linux-amd64" ]]; then
+    if [[ $platform =~ linux/amd64 ]]; then
```
Comment (borkdude) [inline .circleci/script/docker:47]: What does this syntax mean?

Comment (cap10morgan) [reply .circleci/script/docker:47]: actually IIRC there's a simpler substring thingy I can use instead

Comment (borkdude) [reply .circleci/script/docker:47]: Where does the `-` -> `/` change come from?

Comment (cap10morgan) [reply .circleci/script/docker]: changed to substring match

Comment (cap10morgan) [reply .circleci/script/docker]: > Where does the `-` -> `/` change come from?

That should have already been in there, but it comes from the fact that docker platform strings look like `os/arch` while GraalVM and babashka use `os-arch`. There was a time where I was using the hyphenated form in this env var, but that was no longer the case by the time the original PR was merged and I just missed converting this one b/c it was never being run anyway.

Comment (lispyclouds) [reply .circleci/script/docker]: more reasons for this to be in clojure! quite some arcane bash magic!

**Comment by lispyclouds** [comment]
@cap10morgan Im thinking do we need this if check at all? this script is only called once in the ci build process and the value of platform never really changes.

File: .circleci/script/docker [hunk 2]
```diff
@@ -44,9 +44,9 @@ if [ -z "$CIRCLE_PULL_REQUEST" ] && [ "$CIRCLE_BRANCH" = "master" ]; then
     done
 
     # alpine doesn't provide upstream arm64 images yet
-    if [[ $platform == "linux-amd64" ]]; then
+    if [[ $platform == *"linux/amd64"* ]]; then
         tar zxvf  "/tmp/release/babashka-${image_tag}-${platform}-static.tar.gz"
```
Comment (lispyclouds) [inline .circleci/script/docker:48]: we should hardcode the `platform` to `linux-amd64` here right?

**Comment by cap10morgan** [comment]
> @cap10morgan Im thinking do we need this if check at all? this script is only called once in the ci build process and the value of platform never really changes.

I'm happy to take it out if you think it's not needed. All this stuff was platform-parameterized when I found it, so I left it like that from a "first, do no harm" perspective. I'm fine w/ it either way.

**Comment by lispyclouds** [comment]
Yeah lets remove it. We have no other use for the docker script apart from the CI, people building locally anyways use a different Dockerfile and proabably a manual process.

**Review by lispyclouds — APPROVED** [review]

**Pull request merged**